pubsub/src/main/java/com/github/twitch4j/pubsub/domain/SubscriptionData.java

```diff
@@ -0,0 +1,99 @@
+package com.github.twitch4j.pubsub.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
+
+@Data
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SubscriptionData {
+
+    /**
+     * The login name for the user that purchased the subscription
+     */
+    private String userName;
+
+    /**
+     * The display name for the user that purchased the subscription
+     */
+    private String displayName;
+
+    /**
+     * The name of the channel where the subscription took place
+     */
+    private String channelName;
+
+    /**
+     * The id for the user that purchased the subscription
+     */
+    private String userId;
+
+    /**
+     * The id of the channel where the subscription took place
+     */
+    private String channelId;
+
+    /**
+     * RFC 3339 timestamp of when the subscription took place
+     */
+    private String time;
+
+    /**
+     * The type of subscription plan that was purchased;
+     * "Prime" or "1000" or "2000" or "3000"
+     */
+    private String subPlan;
```

Comment:
Should this be converted to an enum

Reply:
Yes, we should use a enum here as we're also using one for subscriptions from the chat module `com.github.twitch4j.chat.enums.SubscriptionPlan`, maybe make a new one in the common module, then we can also switch the chat module to the common enum later on.